src/main/java/org/opensearch/ad/AnomalyDetectorExtension.java

```diff
@@ -39,7 +44,14 @@ public ExtensionSettings getExtensionSettings() {
 
     @Override
     public List<ExtensionRestHandler> getExtensionRestHandlers() {
-        return List.of(new RestCreateDetectorAction());
+        List<ExtensionRestHandler> handler = null;
+        // Get the settings anc clusterService from createComponent
+        try {
+            handler = List.of(new RestCreateDetectorAction(null, null));
```

Comment:
First null is `Settings`, second is cluster service.  
Settings should be available via `ExtensionsRunner.sendEnvironmentSettingsRequest()`.  

BaseExtension createComponents already stores cluster state, just need to actually call that from the ExtensionsRunner at some point, and also add settings to that.

Reply:
No need of settings and clusterService anymore. Removed it :)